### PR TITLE
Update Iceberg from 1.3.1 to 1.4.1

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2049,6 +2049,14 @@
         </dependencies>
     </dependencyManagement>
 
+    <repositories>
```
Comment (findinpath) [inline pom.xml:2052]: Please remove this repository

**Review by findinpath — APPROVED** [review]
LGTM % pom.xml changes

**Comment by ebyhr** [comment]
Please ping after Iceberg 1.4.1 is released officially.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java [hunk 2]
```diff
@@ -382,7 +382,7 @@ private static Metrics loadMetrics(TrinoInputFile file, HiveStorageFormat storag
         InputFile inputFile = new ForwardingInputFile(file);
         return switch (storageFormat) {
             case ORC -> OrcMetrics.fromInputFile(inputFile, METRICS_CONFIG, nameMapping);
-            case PARQUET -> TrinoParquetUtil.fileMetrics(inputFile, METRICS_CONFIG, nameMapping);
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java:385]: cc @raunaqmorarka

Comment (raunaqmorarka) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java:385]: @ebyhr ?

Comment (ebyhr) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java:385]: This change loos fine for me.

File: pom.xml (hunk 1 above)
Comment (nastra) [reply pom.xml:2052]: yes that will be removed once 1.4.1 is officially released

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java [hunk 3]
```diff
@@ -48,7 +48,6 @@ public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalog
         return new NessieIcebergClient(
                 HttpClientBuilder.builder()
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:49]: HttpClientBuilder is deprecated now. Let's migrate to NessieClientBuilder in follow-up.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java [hunk 4]
```diff
@@ -48,7 +48,6 @@ public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalog
...
                         .withUri(icebergNessieCatalogConfig.getServerUri())
-                        .withEnableApiCompatibilityCheck(false)
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:51]: Why did you remove this line? The method was renamed to withApiCompatibilityCheck, not removed.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java [hunk 5]
```diff
@@ -125,10 +127,10 @@ protected void testCreateTableFailure(String expectedExceptionMessage, boolean s
 
         Path metadataDirectory = Path.of(tableLocation, "metadata");
         if (shouldMetadataFileExist) {
-            assertThat(metadataDirectory).as("Metadata file should exist").isDirectoryContaining("glob:**.metadata.json");
+            assertThat(metadataDirectory).as("Metadata file should exist").isDirectoryContaining(METADATA_GLOB);
         }
         else {
-            assertThat(metadataDirectory).as("Metadata file should not exist").isEmptyDirectory();
+            assertThat(metadataDirectory).as("Metadata file should not exist").isDirectoryNotContaining(METADATA_GLOB);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: This is a regression. Why did the test start leaving files in `metadata` directory?

**Comment by ebyhr** [comment]
Please squash commits into one.

**Comment by ebyhr** [comment]
/test-with-secrets sha=1b4c5f8687c2b0e23efa8374def53cf4184340b0

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java (hunk 4 above)
Comment (ajantha-bhat) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:51]: This was added in the pervious Iceberg version bump as a temporary workaround for a bug in older Nessie version. It got fixed in the newer version. Hence, removed it. 

more info: https://github.com/trinodb/trino/pull/17715#discussion_r1213150069

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java (hunk 3 above)
Comment (ajantha-bhat) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:49]: ACK

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java (hunk 4 above)
Comment (ebyhr) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:51]: Thanks for explaining the context. It would be nice to leave such information in the commit body for future readers.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java (hunk 5 above)
Comment (nastra) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: https://github.com/trinodb/trino/pull/19188#issuecomment-1740113189

Comment (ebyhr) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: @nastra Please restore the original behavior or leave TODO comment.

Comment (nastra) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: @ebyhr what kind of TODO would you like me to add here. This behavioral change is expected according to the comments in https://github.com/apache/iceberg/pull/8675.
cc @Fokko

Comment (Fokko) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: This was discussed in the community sync, see https://github.com/apache/iceberg/pull/8599. The new behavior is more defensive in cleaning up files in case of a conflict. This might introduce lingering files but allow it to recover to a certain version, or be able to backtrack what happened.

Comment (ebyhr) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:133]: Then, let's leave a code comment explaining the context.

Comment (nastra) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java:134]: added a comment and mentioned the Iceberg PR

**Review by ebyhr — APPROVED** [review]

**Comment by mosabua** [comment]
There are really no using facing improvements from this and we therefore dont need a release notes entry @ebyhr @nastra @Fokko

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java [hunk 6]
```diff
@@ -4704,7 +4704,7 @@ protected void verifyIcebergTableProperties(MaterializedResult actual)
         assertThat(actual).isNotNull();
         MaterializedResult expected = resultBuilder(getSession())
                 .row("write.format.default", format.name())
-                .build();
+                .row("write.parquet.compression-codec", "zstd").build();
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:4707]: @nastra this doesn't seem right - the parquet compression codec property is relevant only for parquet files (not for avro, orc). 
https://github.com/trinodb/trino/issues/20401

Comment (nastra) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:4707]: we should adress this as part of https://github.com/apache/iceberg/issues/9490
